### PR TITLE
Expose Orchestrator cache to service extensions.

### PR DESCRIPTION
**Pull request by daniilstrata**

### CODE REVIEW

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -16,6 +17,10 @@ type Orchestrator interface {
 	// Session returns the session.
 	Session(opts ...session.SessionOpt) (session.Provider, error)
 
+	// RequestCache returns a cache specifically to only be used by service
+	// extensions.
```
Comment (eliasjf) [inline orchestrator/orchestrator.go:21]:
```suggestion
	// RequestCache returns a cache that can be used to store state across different service 
	// extensions.
```

File: cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// WithTTL can be used to add a Time-To-Live (TTL) to a given piece of data. When
+// this option is set, it is up to the Store implementations to determine how
+// best to honor this option.
+func WithTTL(ttl time.Duration) Option {
+	return func(do *Options) {
+		do.TTL = ttl
+	}
+}
+
+// Constraints contains Constraints for a given piece of data.
```
Comment (eliasjf) [inline cache/cache.go:36]:
```suggestion
// Constraints are the constraints for a Cache.
```

File: cache/cache.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+type Constraints struct {
+	// Name of the cache.
+	Name string
+}
+
+// Constraint is an option to pass to the Cache setters. Option can be used to add
+// additional capabilities or modify the default behavior for how the data is stored.
+type Constraint func(*Constraints)
+
+// WithName is an option that allows the ability to specify the cache name.
```
Comment (eliasjf) [inline cache/cache.go:46]:
```suggestion
// WithName is an option to specify the cache name.
```

File: cache/cache.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+// Constraints contains Constraints for a given piece of data.
+type Constraints struct {
+	// Name of the cache.
+	Name string
+}
+
+// Constraint is an option to pass to the Cache setters. Option can be used to add
+// additional capabilities or modify the default behavior for how the data is stored.
```
Comment (eliasjf) [inline cache/cache.go:43]:
```suggestion
// Constraint allows for customizing the Cache.
```

File: cache/cache.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	TTL time.Duration
+}
+
+// Option is an option to pass to the Cache setters. Option can be used to add
+// additional capabilities or modify the default behavior for how the data is stored.
+type Option func(*Options)
+
+// WithTTL can be used to add a Time-To-Live (TTL) to a given piece of data. When
+// this option is set, it is up to the Store implementations to determine how
+// best to honor this option.
```
Comment (eliasjf) [inline cache/cache.go:29]:
```suggestion
// WithTTL can be used to add a Time-To-Live (TTL) to a given piece of data.
```

File: cache/cache.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
+package cache
+
+import "time"
+
+type Cache interface {
+	// GetBytes returns the []byte for a given key. If the key does not exist, a
+	// ErrNotFound will be returned.
```
Comment (eliasjf) [inline cache/cache.go:7]:
```suggestion
	// GetBytes returns the []byte for a given key. If the key does not exist, an
	// error will be returned.
```

**Review by kewun — APPROVED** [review]
Paired with Daniil

**Pull request merged**